test/Olcs/src/FormService/Form/Lva/People/SoleTrader/ApplicationSoleTraderTest.php

```diff
@@ -48,105 +49,15 @@ class ApplicationSoleTraderTest extends MockeryTestCase
-                function ($serviceLocator, $name, $requestedName) {
-                    $element = new DynamicRadio();
-                    $element->setValueOptions(
-                        [
-                            '1' => 'one',
-                            '2' => 'two',
-                            '3' => 'three'
-                        ]
-                    );
-                    return $element;
-                }
-            );
-
-            $this->serviceManager->get('FormElementManager')->setFactory(
-                'Common\Form\Element\DynamicMultiCheckbox',
-                function ($serviceLocator, $name, $requestedName) {
-                    $element = new DynamicMultiCheckbox();
-                    $element->setValueOptions(
-                        [
-                            '1' => 'one',
-                            '2' => 'two',
-                            '3' => 'three'
-                        ]
-                    );
-                    return $element;
-                }
-            );
-
-        return $this->serviceManager;
-    }
-
-    /**
-     * Added this method for backwards compatibility
-     *
-     * @return \Laminas\ServiceManager\ServiceManager
-     */
-    public static function getRealServiceManager()
-    {
-        $serviceManager = new ServiceManager(new ServiceManagerConfig());
-        $config = include 'config/application.config.php';
-        $serviceManager->setService('ApplicationConfig', $config);
-        $serviceManager->get('ModuleManager')->loadModules();
-        $serviceManager->setAllowOverride(true);
-
-        $mockTranslationLoader = m::mock(TranslationLoader::class);
-        $mockTranslationLoader->shouldReceive('load')->andReturn(['default' => ['en_GB' => []]]);
-        $mockTranslationLoader->shouldReceive('loadReplacements')->andReturn([]);
-        $serviceManager->setService(TranslationLoader::class, $mockTranslationLoader);
-
-        $pluginManager = new LoaderPluginManager($serviceManager);
-        $pluginManager->setService(TranslationLoader::class, $mockTranslationLoader);
-        $serviceManager->setService('TranslatorPluginManager', $pluginManager);
-
-        // Mess up the backend, so any real rest calls will fail
-        $config = $serviceManager->get('Config');
-        $config['service_api_mapping']['endpoints']['backend'] = 'http://some-fake-backend/';
-        $serviceManager->setService('Config', $config);
-
-        return $serviceManager;
-    }
-
     public function setUp(): void
     {
         $this->formHelper = m::mock('\Common\Service\Helper\FormHelperService');
-
-        $this->sm = self::getServiceManager();
+        $this->form = m::mock(Form::class);
 
         /** @var FormServiceManager fsm */
         $this->fsm = m::mock('\Common\FormService\FormServiceManager')->makePartial();
-        $this->fsm->setServiceLocator($this->sm);
+        $this->fsm->allows('getServiceLocator')
+            ->andReturns($this->form);
```

Comment:
I'd not expect a `getServiceLocator` call to return a `Form` instance. Can you double check this please?